boards/BMS/Src/App/states/App_ChargeState.c

```diff
@@ -45,29 +49,31 @@ static void ChargeStateRunOnTick100Hz(struct StateMachine *const state_machine)
             has_charger_faulted = App_Charger_HasFaulted(charger);
         }
 
-        uint8_t    segment = 0U;
-        uint8_t    cell    = 0U;
-        const bool has_reached_max_v =
-            App_Accumulator_GetMaxVoltage(accumulator, &segment, &cell) > MAX_CELL_VOLTAGE_THRESHOLD;
+        const bool has_charging_completed = App_TractiveSystem_GetCurrent(ts) <= CURRENT_AT_MAX_CHARGE;
 
         App_CanTx_SetPeriodicSignal_CHARGER_DISCONNECTED_IN_CHARGE_STATE(can_tx, is_charger_disconnected);
         App_CanTx_SetPeriodicSignal_CHARGER_FAULT_DETECTED(can_tx, has_charger_faulted);
-        App_CanTx_SetPeriodicSignal_IS_CHARGING_COMPLETE(can_tx, has_reached_max_v);
+        App_CanTx_SetPeriodicSignal_IS_CHARGING_COMPLETE(can_tx, has_charging_completed);
         App_CanTx_SetPeriodicSignal_CHARGING_EXT_SHUTDOWN_OCCURRED(can_tx, has_external_shutdown_occurred);
 
-        if (is_charger_disconnected || has_charger_faulted || has_reached_max_v || has_external_shutdown_occurred)
+        // If the current indicates charging is complete, the shutdown loop is initiated, charging is disabled over CAN
+        // or charger is disconnected Go back to init state
+        if (has_charging_completed || has_external_shutdown_occurred || !charge_over_can || is_charger_disconnected)
         {
-            App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
+            App_SharedStateMachine_SetNextState(state_machine, App_GetInitState());
```

Comment:
we need to ensure that AIR+ is opened again. it should either go in Charge RunOnExit or Init RunOnEntry. I think I prefer Init RunOnEntry because it will also apply to any other state without a transition through fault. Its nice to guarentee that AIR+ is open in Init State

Reply:
Would this also cause cycling? If charging complete go to init, but charger could still be plugged in, so precharge again and back to charge state, etc. etc.

Reply:
Could be best to add it in both places tbh. Charge RunOnExit for clarity that contactors open after charger state, and Init RunOnEntry to be redundant.

Reply:
Sounds good, should be added in both places now